ext/hathach/uart.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+* Copyright (c) 2020, Erik Henriksson
+* Copyright (c) 2020, Niklas Hauser
+*
+* This file is part of the modm project.
+*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <modm/architecture/interface/uart.hpp>
+#include "tusb.h"
+
+namespace modm::platform
+{
+
+template< uint8_t Port=0 >
```

Comment:
Nit: Making this a field will save a whole bunch of binary size as C++ does not need to replicate the whole class for each instance.

I doubt it affects performance much as loading a field vs a constant is much of a difference.

Reply:
Agreed.

Reply:
Meh for now. Just too comfortable to write `UsbUart0` ;-P I'll revisit this later.

Reply:
`UsbUart uart(0);`?

Reply:
Considering that with 4 endpoints you can only have two CDC ports, I think the overhead is fine. This makes them behave like `Uart` ports.